### PR TITLE
Feature/imagegallery

### PR DESCRIPTION
**Pull request by pusztaienike**

### CODE REVIEW

File: examples/sn-react-imagegallery/package.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    "typescript": "^3.7.2"
+  },
+  "dependencies": {
+    "@material-ui/core": "^4.6.1",
+    "@material-ui/icons": "^4.5.1",
+    "@sensenet/client-core": "^2.2.4",
+    "@sensenet/default-content-types": "^2.0.3",
+    "@sensenet/hooks-react": "^1.0.3",
+    "moment": "^2.24.0",
+    "react": "^16.12.0",
```
Comment (B3zo0) [inline examples/sn-react-imagegallery/package.json]: Please remove the duplicate react, react-dom packages.

File: examples/sn-react-imagegallery/public/index.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta name="description" content="Sensenet Image gallery example app" />
```
Comment (B3zo0) [inline examples/sn-react-imagegallery/public/index.html]:
```suggestion
    <meta name="description" content="sensenet Image gallery example app" />
```

File: examples/sn-react-imagegallery/public/index.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    <title>Sensenet Image gallery example app</title>
```
Comment (B3zo0) [inline examples/sn-react-imagegallery/public/index.html]:
```suggestion
    <title>sensenet Image gallery example app</title>
```

File: examples/sn-react-imagegallery/src/app.tsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
+import React, { useEffect, useState } from 'react'
+import { ConstantContent } from '@sensenet/client-core'
+import moment from 'moment'
+import { CssBaseline, Slide } from '@material-ui/core'
+import { TransitionProps } from '@material-ui/core/transitions'
+import { makeStyles } from '@material-ui/core/styles'
+import { Image, User } from '@sensenet/default-content-types'
+import snLogo from './assets/sensenet_logo_transparent.png'
+import { useRepository } from './hooks/use-repository'
```
Comment (B3zo0) [inline examples/sn-react-imagegallery/src/app.tsx]: Please refactor this to use `useRepository` from @sensenet/hooks-react in all components.

File: examples/sn-react-imagegallery/test/repository-provider.test.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { shallow } from 'enzyme'
+import React from 'react'
+import { RepositoryProvider } from '../src/context/repository-provider'
+
+describe('Repository Context Provider', () => {
+  it('Matches snapshot', () => {
```
Comment (B3zo0) [inline examples/sn-react-imagegallery/test/repository-provider.test.tsx]: Please remove this test.

File: examples/sn-react-imagegallery/src/context/repository-provider.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
+import React, { createContext, useEffect, useState } from 'react'
+import { FormsAuthenticationService, LoginState, Repository } from '@sensenet/client-core'
+import { RepositoryConfiguration } from '@sensenet/client-core/dist/Repository/RepositoryConfiguration'
+import { LoginForm } from '../components/login-form'
+import { FullScreenLoader } from '../components/full-screen-loader'
+
+/**
+ * React context that stores a sensenet Repository object
+ */
+export const RepositoryContext = createContext(new Repository())
```
Comment (B3zo0) [inline examples/sn-react-imagegallery/src/context/repository-provider.tsx]: Use RepositoryContext from @sensenet/hooks-react

File: examples/sn-react-imagegallery/src/hooks/use-current-user.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from 'react'
+import { User } from '@sensenet/default-content-types'
+import { useRepository } from './use-repository'
+
+/**
+ * Custom hook that will return with the current user.
+ */
+export const useCurrentUser = () => {
+  const repo = useRepository()
```
Comment (B3zo0) [inline examples/sn-react-imagegallery/src/hooks/use-current-user.ts]: You can remove this file if useCurrentUser is not used.

File: examples/sn-react-imagegallery/src/hooks/use-repository.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import { useContext } from 'react'
+import { RepositoryContext } from '../context/repository-provider'
+
+/**
+ * Custom hook that will return with a Repository object
+ */
+export const useRepository = () => {
+  return useContext(RepositoryContext)
```
Comment (B3zo0) [inline examples/sn-react-imagegallery/src/hooks/use-repository.ts]: Please remove this file.

File: examples/sn-react-imagegallery/test/DotsMobileStepper.test.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+describe('handleNext', () => {
+  const mock = {
+    imageIndex: 1,
+    steppingFunction: jest.fn(),
+    imageListLenght: 1,
+  }
+  it('Matches snapshot', () => {
+    const wrapper = mount(<DotsMobileStepper {...mock} />)
+    expect(wrapper).toMatchSnapshot()
```
Comment (B3zo0) [inline examples/sn-react-imagegallery/test/DotsMobileStepper.test.tsx]: Remove this snapshot test please.

File: examples/sn-react-imagegallery/src/components/login-form.tsx [hunk 10]
```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react'
+import { Button, Divider, Paper, TextField, Typography } from '@material-ui/core'
+import { lastRepositoryKey } from '../context/repository-provider'
+
+export interface LoginFormProps {
+  /**
+   * Callback that will be called when the user clicks on the Login button
```
Comment (herflis) [inline examples/sn-react-imagegallery/src/components/login-form.tsx]:
```suggestion
   * Callback that will be called when the user clicks the Login button
```

File: examples/sn-react-imagegallery/src/components/login-form.tsx [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+  onLogin: (username: string, passowrd: string, repository: string) => void
+
+  /**
+   * An optional error message
+   */
+  error?: string
+}
+
+/**
+ * Login component for Sensenet repositories
```
Comment (herflis) [inline examples/sn-react-imagegallery/src/components/login-form.tsx]:
```suggestion
 * Login component for sensenet repositories
```

File: examples/sn-react-imagegallery/src/components/AdvancedGridList.tsx [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+  icon: {
+    color: 'white',
+  },
+  titleBar: {
+    background: 'linear-gradient(to bottom, rgba(0,0,0,0.7) 0%, rgba(0,0,0,0.3) 70%, rgba(0,0,0,0) 100%)',
+  },
+}))
+
+/**
+ * Determines which image should be full width
```
Comment (herflis) [inline examples/sn-react-imagegallery/src/components/AdvancedGridList.tsx]:
```suggestion
 * Determines which image should be full width sized
```

File: examples/sn-react-imagegallery/test/AdvancedGridList.test.tsx [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+    openFunction: jest.fn(),
+    imgList: images as any[],
+  }
+  it('Matches snapshot', () => {
+    const wrapper = shallow(<AdvancedGridList {...testprops} />)
+    expect(wrapper).toMatchSnapshot()
+  })
+
+  describe('pickTile', () => {
+    it('should return 2', () => {
```
Comment (herflis) [inline examples/sn-react-imagegallery/test/AdvancedGridList.test.tsx]:
```suggestion
    it('should return 2 when the given number is divisible by three', () => {
```

File: examples/sn-react-imagegallery/test/app.test.tsx [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+import { FullScreenDialog } from '../src/components/FullScreenDialog'
+import { emptyimages, images } from './mocks/images'
+import moment = require('moment')
+
+describe('App Layout', () => {
+  it('Matches snapshot', () => {
+    const l = shallow(<App />)
+    expect(l).toMatchSnapshot()
+  })
+  it('Transition snapshot', () => {
```
Comment (herflis) [inline examples/sn-react-imagegallery/test/app.test.tsx]:
```suggestion
  it('matches Transition snapshot', () => {
```

File: examples/sn-react-imagegallery/test/app.test.tsx [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+    const l = shallow(<Transition />)
+    expect(l).toMatchSnapshot()
+  })
+  it('getselectedImage with result', async () => {
```
Comment (herflis) [inline examples/sn-react-imagegallery/test/app.test.tsx]:
```suggestion
  it('returns the selected image', async () => {
```

File: examples/sn-react-imagegallery/test/app.test.tsx [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+      .prop('openedImg')
+    expect(obj.imgTitle).toBe(images[0].DisplayName)
+    expect(obj.imgPath).toBe(repository.configuration.repositoryUrl + images[0].Path)
+    expect(obj.imgAuthorAvatar).toBe(images[0].CreatedBy.Avatar.Url)
+    expect(obj.imgDescription).toBe(images[0].Description)
+    expect(obj.imgCreationDate).toBe(moment(images[0].CreationDate).format('YYYY-MM-DD HH:mm:ss'))
+    expect(obj.imgAuthor).toBe(images[0].CreatedBy.DisplayName)
+    expect(obj.imgSize).toBe(`${(images[0].Size / 1024 / 1024).toFixed(2)} MB`)
+  })
+  it('getselectedImage without result', async () => {
```
Comment (herflis) [inline examples/sn-react-imagegallery/test/app.test.tsx]:
```suggestion
  it('returns without a result when getSelectedImage is called', async () => {
```

File: examples/sn-react-imagegallery/test/login-form.test.tsx [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+import TextField from '@material-ui/core/TextField'
+import { act } from 'react-dom/test-utils'
+import { LoginForm } from '../src/components/login-form'
+
+describe('LoginForm', () => {
+  const testprop = {
+    onLogin: jest.fn(),
+    error: 'test',
+  }
+  it('LoginForm snapshot', () => {
```
Comment (herflis) [inline examples/sn-react-imagegallery/test/login-form.test.tsx]:
```suggestion
  it('matches LoginForm snapshot', () => {
```

File: examples/sn-react-imagegallery/test/login-form.test.tsx [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+    const wrapper = shallow(<LoginForm {...testprop} />)
+    expect(wrapper).toMatchSnapshot()
+  })
+  it('Username change test', async () => {
```
Comment (herflis) [inline examples/sn-react-imagegallery/test/login-form.test.tsx]:
```suggestion
  it('changes the username', async () => {
```

File: examples/sn-react-imagegallery/test/login-form.test.tsx [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+    })
+    expect(
+      wrapper
+        .update()
+        .find(TextField)
+        .first()
+        .props().value,
+    ).toEqual('testusername')
+  })
+  it('Repository change test', async () => {
```
Comment (herflis) [inline examples/sn-react-imagegallery/test/login-form.test.tsx]:
```suggestion
  it('changes the repository', async () => {
```

**Pull request merged**